packages/app/src/components/404/Custom404Page.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { Box, Button, Divider, Grid, Typography, makeStyles } from '@material-ui/core';
+import { Content, Page } from '@backstage/core-components';
+import { SearchBar, SearchContextProvider, SearchPagination } from '@backstage/plugin-search-react';
+import { searchPlugin } from '@backstage/plugin-search';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { searchResultCustomList } from '../search/SearchResultCustomList';
+import { useRouteRef } from '@backstage/core-plugin-api';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+
+const useStyles = makeStyles(theme => ({
+  title: {
+    gap: theme.spacing(1),
+    display: 'grid',
+    alignItems: 'center',
+    gridTemplateColumns: '1fr auto',
+    '&> button': {
+      marginTop: theme.spacing(1),
+    },
+  },
+  input: {
+    flex: 1,
+  },
+  link: {
+    color: theme.palette.primary.main,
+  },
+  button: {
+    '&:hover': {
+      background: 'none',
+    },
+  },
+}));
+
+const rootRouteRef = searchPlugin.routes.root;
+
+export function Custom404Page() {
+  const classes = useStyles();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const searchRootRoute = useRouteRef(rootRouteRef)();
+  const searchBarRef = useRef<HTMLInputElement | null>(null);
+
+  // grab the url slug
+  const pagePath = location.pathname.split('/').pop();
+
+  const preFiltered = {
+    term: pagePath?.replaceAll('-', ' ')?.replaceAll('_', ' ') ?? '',
+    types: [],
+    filters: {},
+  };
+
+  useEffect(() => {
+    searchBarRef?.current?.focus();
+  });
+
+  // This handler is called when "enter" is pressed
+  const handleSearchBarSubmit = useCallback(() => {
+    // Using ref to get the current field value without waiting for a query debounce
+    const query = searchBarRef.current?.value ?? '';
+    navigate(`${searchRootRoute}?query=${query}`);
+  }, [navigate, searchBarRef]);
+
+  return (
+    <Page themeId="home">
+      <Content>
+        <SearchContextProvider initialState={preFiltered}>
+          <Box padding="calc(2.1rem - 24px) 9%">
+            <Typography variant="h1">Page not found</Typography>
+            <Divider orientation="horizontal" variant="middle" />
+            <Box pt={3}>
+              <Typography variant="h6">
+                <Link className={classes.link} to="#" onClick={() => navigate(-1)}>Go back</Link> or&nbsp;
```

Comment:
I defer to Shivangani here as the content expert, but my thought is

- we don't need 'go back' people know how to use a back button?
-Move the rest of to the bottom after the search results, as they would need to read through the search results before knowing if they didn't find what they were looking for

Reply:
I feel like a "go back" button functions differently for mobile app form, whereas desktop/laptop has a steady browser button. There could be a potential when pressing the back button on a phone, it exits the app altogether.

If that's not the case, then no misalignment with JR's comment.